server/src/main/java/org/opensearch/search/aggregations/MultiBucketConsumerService.java

```diff
@@ -133,7 +133,7 @@ public static class MultiBucketConsumer implements IntConsumer {
 
         // aggregations execute in a single thread for both sequential
         // and concurrent search, so no atomic here
-        private int count;
+        private final LongAdder count;
```

Comment:
Does the comment above still apply? You are essentially making it atomic.

Reply:
Fixed in the latest revision